src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/PoPUserOTPService.kt

```diff
@@ -44,11 +44,9 @@ class PoPUserOTPService(
   fun createPoPUserOTP(prisoner: PrisonerEntity): PoPUserOTPEntity {
     val now = LocalDateTime.now()
     val popUserOTPExists = popUserOTPRepository.findByPrisoner(prisoner)
-    // val secureRandom = SecureRandom()
     // For now OTP generated is in 6 digits, for 8 digits the below value should be 99999999
     val otp = SecureRandom.getInstanceStrong().nextLong(999999)
-    val otpValue = String.format("%06d", otp).toLong()
-    SecureRandom.getInstanceStrong()
+    val otpValue = String.format("%06d", otp).reversed().toLong()
```

Comment:
are you sure this will generate 6 digits? we are seeing some instances of 5

Reply:
modified code and added test unit to generate 1000 otp